Source/WebCore/rendering/RenderGrid.cpp

```diff
@@ -699,6 +699,16 @@ void RenderGrid::computeIntrinsicLogicalWidths(LayoutUnit& minLogicalWidth, Layo
 
     computeTrackSizesForIndefiniteSize(algorithm, GridTrackSizingDirection::ForColumns, gridLayoutState, &minLogicalWidth, &maxLogicalWidth);
 
+    if (isMasonry(GridTrackSizingDirection::ForColumns)) {
+        computeTrackSizesForIndefiniteSize(algorithm, GridTrackSizingDirection::ForRows, gridLayoutState, &minLogicalWidth, &maxLogicalWidth);
+
+        unsigned gridAxisTracksBeforeAutoPlacement = currentGrid().numTracks(GridTrackSizingDirection::ForRows);
+        m_masonryLayout.performMasonryPlacement(gridAxisTracksBeforeAutoPlacement, GridTrackSizingDirection::ForColumns);
+
+        minLogicalWidth = m_masonryLayout.gridContentSize();
+        maxLogicalWidth = m_masonryLayout.gridContentSize();
```

Comment:
I think we should add a comment explaining why we think it is ok to use the grid content size for both `minLogicalWidth` and `maxLogicalWidth` in this case